catroid/src/main/java/org/catrobat/catroid/content/actions/WriteListOnDeviceAction.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.content.actions
+
+import android.util.Log
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.formulaeditor.UserList
+import org.catrobat.catroid.io.DeviceListAccessor
+import org.catrobat.catroid.io.DeviceUserDataAccessor
+import org.catrobat.catroid.utils.CoroutineAsyncTask
+import java.io.IOException
+
+class WriteListOnDeviceAction : AsynchronousAction() {
+    private var userList: UserList? = null
+    private var writeActionFinished = false
+
+    override fun act(delta: Float): Boolean {
+        return if (userList == null) {
+            true
+        } else super.act(delta)
+    }
+
+    override fun initialize() {
+        writeActionFinished = false
+        WriteTask().execute(userList)
+    }
+
+    override fun isFinished(): Boolean = writeActionFinished
+
+    fun setUserList(userList: UserList?) {
+        this.userList = userList
+    }
+
+    private inner class WriteTask : CoroutineAsyncTask<UserList?, Void?, Void?>() {
+        override fun onPostExecute(result: Void?) {
+            writeActionFinished = true
+        }
+
+        override fun doInBackground(userList: Array<out UserList?>): Void? {
+            val projectDirectory = ProjectManager.getInstance().currentProject.directory
+            val accessor: DeviceUserDataAccessor = DeviceListAccessor(projectDirectory)
+            for (list in userList) {
+                try {
+                    accessor.writeUserData(list)
```

Comment:
Please move the exception handling inside of accessor.writeUserData(list) and then remove the try catch blocks where they are not necesessary anymore.